### PR TITLE
Taboola actions implementation

### PR DESCRIPTION
**Pull request by eden-ohana**
This PR adds Taboola as a destination.

##### Authentication:
* OAuth Managed Scheme:
    - Implements OAuth managed authentication with client_id and client_secret fields.
* Refresh Access Token:
    - Defines a method to refresh the access token by making a POST request to the Taboola OAuth token endpoint using client credentials.
* Extend Request:
    - Adds authorization headers with the bearer token for outgoing requests.

##### Create Audience:
* Validation:
    - Validates required fields (audienceName and account_id).
* Request:
    - Sends a request to Taboola to create an audience with the specified name, TTL, and exclusion setting.
* Response:
    - Returns the external ID of the created audience.

##### Actions:
* syncAudience:
    - Integrates the syncAudience action, allowing the synchronization of audience data with [Taboola](https://developers.taboola.com/backstage-api/reference/add-or-remove-users).
    - Supports batch by default.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by eden-ohana** [comment]
@joe-ayoub-segment can you please add reviewers?

File: packages/destination-actions/src/destinations/taboola-actions/syncAudience/index.ts [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+      required: true,
+      unsafe_hidden: true
+    }
+  },
+  perform: (request, { payload, audienceSettings }) => {
+    if (!audienceSettings) {
+      throw new IntegrationError('Bad Request: no audienceSettings found.', 'INVALID_REQUEST_DATA', 400)
+    }
+
+    const action = payload.traits_or_props[payload.segment_computation_key] as boolean //TODO action null?
```
Comment (eden-ohana) [inline packages/destination-actions/src/destinations/taboola-actions/syncAudience/index.ts:152]: Could `segment_computation_key` be null?

File: packages/destination-actions/src/destinations/taboola-actions/syncAudience/index.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    if (clusters.length === 0) {
+      throw new IntegrationError(
+        "Either 'Email address' or 'Mobile Device ID' must be provided in the payload",
+        'MISSING_REQUIRED_FIELD',
+        400
+      )
+    }
+
+    const action = payloads[0].traits_or_props[payloads[0].segment_computation_key] as boolean
```
Comment (eden-ohana) [inline packages/destination-actions/src/destinations/taboola-actions/syncAudience/index.ts:196]: Making sure that when a batch operation is triggered, all events will have the same `segment_computation_key`.
To know whether to add or to remove all the users.

**Comment by joe-ayoub-segment** [comment]
Thanks for the PR @eden-ohana . I'll schedule for review. It might not be until next week as I have some more PTO later this week.

**Comment by joe-ayoub-segment** [comment]
PR deployed.

**Pull request merged**